func_view test fails on Appveyor with access violation

My func_view test from PR #4150 for #4083 works on Windows locally (ran 20x in a row on each of 2 different 1909 machines) but on Appveyor it has a mysterious crash:

```
[00:24:03] 205: Running postcmd |C:/projects/dynamorio/build/build_debug-internal-64/clients/bin64/drcachesim.exe;-indir;C:/projects/dynamorio/build/build_debug-internal-64/suite/tests/tool.drcacheoff.func_view.common.fib.exe.07132.7209.dir;-simulator_type;func_view|
[00:24:03] 205: CMake Error at C:/projects/dynamorio/suite/tests/runmulti.cmake:106 (message):
[00:24:03] 205:   *** postcmd failed (Access violation): ***
[00:24:03] 205: 
[00:24:03] 205: Call Stack (most recent call first):
[00:24:03] 205:   C:/projects/dynamorio/suite/tests/runmulti.cmake:118 (process_cmdline)
[00:24:03] 205: 
[00:24:03] 205: 
[00:24:03] 205/239 Test #205: code_api|tool.drcacheoff.func_view ...........................***Failed   14.80 sec
```

For now I'm disabling the test on Windows.

--- a/core/stats.h
+++ b/core/stats.h
@@ -1,5 +1,5 @@
 /* **********************************************************
- * Copyright (c) 2017 Google, Inc.  All rights reserved.
+ * Copyright (c) 2017-2020 Google, Inc.  All rights reserved.
  * Copyright (c) 2004-2009 VMware, Inc.  All rights reserved.
  * **********************************************************/
 
@@ -129,10 +129,11 @@ typedef struct {
 
 extern timestamp_t kstat_ignore_context_switch;
 
-#    define KSTAT_THREAD_NO_PV_START(dc)                                       \
-        do {                                                                   \
-            dcontext_t *cur_dcontext = (dc);                                   \
-            if (cur_dcontext != NULL && cur_dcontext->thread_kstats != NULL) { \
+#    define KSTAT_THREAD_NO_PV_START(dc)                                   \
+        do {                                                               \
+            dcontext_t *cur_dcontext = (dc);                               \
+            if (cur_dcontext != NULL && cur_dcontext != GLOBAL_DCONTEXT && \
+                cur_dcontext->thread_kstats != NULL) {                     \
                 kstat_stack_t *ks = &cur_dcontext->thread_kstats->stack_kstats;
 
 #    define KSTAT_THREAD_NO_PV_END() \

--- a/suite/tests/CMakeLists.txt
+++ b/suite/tests/CMakeLists.txt
@@ -3076,12 +3076,9 @@ endif ()
       torunonly_drcacheoff(view ${ci_shared_app} ""
         "@-simulator_type@view@-sim_refs@16384" "")
 
-      # TODO i#4155: This passes locally but is failing on Appveyor.
-      if (NOT WIN32)
-        set(tool.drcacheoff.func_view_full_run ON) # Fails on Windows if truncated.
-        torunonly_drcacheoff(func_view common.fib "-record_function fib|1"
-          "@-simulator_type@func_view" "only_5")
-      endif ()
+      set(tool.drcacheoff.func_view_full_run ON) # Fails on Windows if truncated.
+      torunonly_drcacheoff(func_view common.fib "-record_function fib|1"
+        "@-simulator_type@func_view" "only_5")
 
       if (LINUX AND X86 AND X64 AND HAVE_RSEQ)
         torunonly_drcacheoff(rseq linux.rseq "" "@-test_mode" "")

